translates: composition-vs-inheritance (#32)
reactjs/nl.reactjs.org: content/docs/composition-vs-inheritance.md

---
id: composition-vs-inheritance
title: Compositie Versus Overerving
permalink: docs/composition-vs-inheritance.html
redirect_from:
  - "docs/multiple-components.html"
prev: lifting-state-up.html
next: thinking-in-react.html
---

> React heeft een krachtig compositie model, en we bevelen aan om compositie te gebruiken in plaats van overerving om code tussen componenten te herbruiken.

Reviewer: Als je "tussen" gebruikt lijkt het er in het Nederlands iets dat letterlijk tussen componenten wordt gebruikt. Maar dat wordt hier niet bedoeld.

Blijkbaar is in het engels "reuse between" goed taal gebruik. In het nederlands lijkt het lastiger, maar de volgende mogelijkheden lijken beter:

- hergebruik door 
- hergebruik onder
- hergebruik voor meer

### Voorgestelde wijzigingen
tussen component => voor meer componenten
te herbuiken => te hergebruiken (herbuiken is minder gangbaar)

```suggestion
React heeft een krachtig compositie model, en we bevelen aan om compositie te gebruiken in plaats van overerving om code voor meer componenten te hergebruiken.
```

### De tekst wordt dan
React heeft een krachtig compositie model, en we bevelen aan om compositie te gebruiken in plaats van overerving om code voor meer componenten te hergebruiken.

> In deze sectie zullen we een paar problemen overwegen waarbij ontwikkelaars die beginnend zijn met React vaak grijpen naar overerving, en tonen hoe we ze kunnen oplossen met compositie.

Reviewer: Keuze tussen "sectie" en "hoofdstuk" moet we ergens denk ik afstemmen.
Bij "tonen" denk ik maar aan letterlijk iets voor iemands ogen houden.
Ik denk dat "beschouwen" gangbaarder is.

### Voorgestelde wijzigingen
sectie -> hoofdstuk?
tonen -> laten we zien?
overwegen -> beschouwen

```suggestion
In dit hoofdstuk zullen we een paar problemen beschouwen waarbij ontwikkelaars die beginnend zijn met React vaak grijpen naar overerving, en laten we zien hoe we ze kunnen oplossen met compositie.
```

### De tekst wordt dan
In dit hoofdstuk zullen we een paar problemen beschouwen waarbij ontwikkelaars die beginnend zijn met React vaak grijpen naar overerving, en laten we zien hoe we ze kunnen oplossen met compositie.

## Insluiting {#containment}

> Sommige componenten weten niet op voorhand welke hun kinderen gaan zijn. Dit is zeer gebruikelijk voor componenten zoals `Sidebar` of `Dialog` die generieke "dozen" representeren.

Reviewer: "especially" betekent niet hetzelfde als "zeer":
[especially](https://nl.bab.la/woordenboek/engels-nederlands/especially)

Ik vind "gebruikelijk bij" iets beter dan "gebruikelijk voor", maar kan het niet goed onderbouwen.

Ik zou "boxes" niet vertalen.

Vertegenwoordigen is iets gangbaarder dan representeren. Representeren voelt iets meer aan als "zich voordoen als" terwijl ze het wel gewoon zijn.

### Voorgestelde wijzigingen
welke hun kinderen => wie hun children
zeer gebruikelijk voor => vooral gebruikelijk bij
"dozen" => "boxes"
representeren => vertegenwoordigen
gaan zijn => worden

```suggestion
Sommige componenten weten niet op voorhand wie hun kinderen worden. Dit is vooral gebruikelijk bij componenten zoals `Sidebar` of `Dialog` die generieke "boxes" vertegenwoordigen.
```

### De tekst wordt dan
Sommige componenten weten niet op voorhand wie hun kinderen worden. Dit is vooral gebruikelijk bij componenten zoals `Sidebar` of `Dialog` die generieke "boxes" vertegenwoordigen.

> We raden aan dat zulke componenten de speciale `children` prop gebruiken om elementen die als kinderen van de component worden meegegeven rechtstreeks naar hun uitvoer door te geven:

Reviewer: ### Voorgestelde wijzigingen
elementen die als kinderen van de component worden meegegeven => child elementen

```suggestion
We raden aan dat zulke componenten de speciale `children` prop gebruiken om child elementen rechtstreeks naar hun uitvoer door te geven:
```

### De tekst wordt dan
We raden aan dat zulke componenten de speciale `children` prop gebruiken om child elementen rechtstreeks naar hun uitvoer door te geven:

}
```

> Dit laat andere componenten toe om willekeurige kinderen door te geven door ze in de JSX te nesten:

Reviewer: ### Voorgestelde wijzigingen
laat andere componenten toe => staat ander componenten toe
kinderen => children

```suggestion
Dit staat andere componenten toe om willekeurige children door te geven door ze in de JSX te nesten:
```

### De tekst wordt dan
Dit staat andere componenten toe om willekeurige children door te geven door ze in de JSX te nesten:

}
```

**[Probeer het op CodePen](https://codepen.io/gaearon/pen/ozqNOV?editors=0010)**

> Eender wat binnen de `<FancyBorder>` JSX tag staat zal doorgegevn worden in de `FancyBorder` component als een `children` prop. Aangezien `FancyBorder` `{props.children}` rendert binnen een `<div>` zullen de doorgegevn elementen verschijnen in de finale uitvoer.

Reviewer: ### Voorgestelde wijzigingen
Eender wat binnen => Iets wat binnen (eender is beetje oubollig)
zal doorgegevn worden in de => wordt doorgegeven aan het
zullen => , zullen (komma voor zullen)
finale => uiteindelijke
doorgegevn => doorgegegeven

```suggestion
Iets wat binnen de `<FancyBorder>` JSX tag staat wordt doorgegeven aan het `FancyBorder` component als een `children` prop. Aangezien `FancyBorder` `{props.children}` rendert binnen een `<div>`, zullen de doorgegeven elementen verschijnen in de uiteindelijke uitvoer.
```

### De tekst wordt dan
Iets wat binnen de `<FancyBorder>` JSX tag staat wordt doorgegeven aan het `FancyBorder` component als een `children` prop. Aangezien `FancyBorder` `{props.children}` rendert binnen een `<div>`, zullen de doorgegeven elementen verschijnen in de uiteindelijke uitvoer.

### Oorspronkelijk
Eender wat binnen de `<FancyBorder>` JSX tag staat zal doorgegevn worden in de `FancyBorder` component als een `children` prop. Aangezien `FancyBorder` `{props.children}` rendert binnen een `<div>` zullen de doorgegevn elementen verschijnen in de finale uitvoer.

> Alhoewel het minder gebruikelijk is, soms moet je meerdere "gaten" in een component hebben. In zulke gevallen mag je je eigen conventie hanteren in de plaats van `children`:

Reviewer: ### Voorgestelde wijzigingen
soms ... hebben => kan je soms meerdere "gaten" in een component nodig hebben
hanteren => bedenken

```suggestion
Alhoewel het minder gebruikelijk is, kan je soms meerdere "gaten" in een component nodig hebben. In zulke gevallen mag je je eigen conventie bedenken in de plaats van `children`:
```

### De tekst wordt dan
Alhoewel het minder gebruikelijk is, kan je soms meerdere "gaten" in een component nodig hebben. In zulke gevallen mag je je eigen conventie bedenken in de plaats van `children`:

### Oorspronkelijk
Alhoewel het minder gebruikelijk is, soms moet je meerdere "gaten" in een component hebben. In zulke gevallen mag je je eigen conventie hanteren in de plaats van `children`:

}
```

[**Probeer het op CodePen**](https://codepen.io/gaearon/pen/gwZOJp?editors=0010)

> React elementen zoals `<Contacts />` en `<Chat />` zijn maar objecten, dus je kan ze als props doorgeven zoals alle andere gegevens. Deze aanpak kan je van in andere bibliotheken bekend voorkomen als "slots" maar in React zijn er geen beperkingen met betrekking tot wat je als props kunt doorgeven.

Reviewer: ### Voorgestelde wijzigingen
kan => kunt
alle andere gegevens => willekeurige andere gegevens
kan ... "slots" => kan je bekend voorkomen uit andere bibliotheken als "slots"

```suggestion
React elementen zoals `<Contacts />` en `<Chat />` zijn maar objecten, dus je kunt ze als props doorgeven zoals willekeurige andere gegevens. Deze aanpak kan je bekend voorkomen uit andere bibliotheken als "slots", maar in React zijn er geen beperkingen met betrekking tot wat je als props kunt doorgeven.
```

### De tekst wordt dan
React elementen zoals `<Contacts />` en `<Chat />` zijn maar objecten, dus je kunt ze als props doorgeven zoals willekeurige andere gegevens. Deze aanpak kan je bekend voorkomen uit andere bibliotheken als "slots", maar in React zijn er geen beperkingen met betrekking tot wat je als props kunt doorgeven.

## Specializatie {#specialization}

> Soms denken we over componenten als "speciale gevallen" van andere componenten. Bijvoorbeeld, We zouyden kunnen zeggen dat een `WelcomeDialog` een speciaal geval van `Dialog` is.

Reviewer: ### Voorgestelde wijzigingen
zouyden => zouden
Bijvoorbeeld, we (kleine letter w)

```suggestion
Soms denken we over componenten als "speciale gevallen" van andere componenten. Bijvoorbeeld, we zouden kunnen zeggen dat een `WelcomeDialog` een speciaal geval van `Dialog` is.
```
### De tekst wordt dan
Soms denken we over componenten als "speciale gevallen" van andere componenten. Bijvoorbeeld, we zouden kunnen zeggen dat een `WelcomeDialog` een speciaal geval van `Dialog` is.

> In React kan dit ook doormiddel van compositie, waar een "specifiekere" component een "generiekere" rendert en configureert met props:

Reviewer: ### Voorgestelde wijzigingen
kan dit door middel van => wordt dit bereikt door?
"specifiekere" => "specifieker"
"generiekere" => "generieker" component 
doormiddel => door middel

```suggestion
In React kan dit ook worden bereikt door middel van compositie, waar een "specifieker" component een "generieker" component rendert en configureert met props:
```

### De tekst wordt dan
In React kan dit ook worden bereikt door middel van compositie, waar een "specifieker" component een "generieker" component rendert en configureert met props:

}
```

[**Probeer het op CodePen**](https://codepen.io/gaearon/pen/gwZbYa?editors=0010)

> ## Dus Wat Met Overerving? {#so-what-about-inheritance}

Reviewer: ### Voorgestelde wijziging
Dus Hoe Zit Het Met Overerving

```suggestion
## Dus Hoe Zit Het Met Overerving? {#so-what-about-inheritance}
```

### De tekst wordt dan
Dus Hoe Zit Het Met Overerving?

Bij Facebook gebruiken we React in duizenden componenten en we hebben nog geen enkel geval gevonden waar we zouden aanraden om een componenten hierarchie met overerving te gebruiken.

> Props en compositie geven je alle flexibiliteit die je nodig hebt om het uitzicht en het gedrag van een component te customizen op een expliciete en veilige manier. Onthoud dat componenten arbitraire props kunnen aannemen, waaronder primitieve waarden, React elementen of functies.

Reviewer: ### Voorgestelde wijzigingen
uitzicht => uiterlijk
te customizen => aan te passen
aannemen => accepteren?
arbitraire => willekeurige

```suggestion
Props en compositie geven je alle flexibiliteit die je nodig hebt om het uiterlijk en het gedrag van een component aan te passen op een expliciete en veilige manier. Onthoud dat componenten willekeurige props kunnen aannemen, waaronder primitieve waarden, React elementen of functies.
```

### De tekst wordt dan
Props en compositie geven je alle flexibiliteit die je nodig hebt om het uiterlijk en het gedrag van een component aan te passen op een expliciete en veilige manier. Onthoud dat componenten willekeurige props kunnen aannemen, waaronder primitieve waarden, React elementen of functies.

### Oorspronkelijk
Props en compositie geven je alle flexibiliteit die je nodig hebt om het uitzicht en het gedrag van een component te customizen op een expliciete en veilige manier. Onthoud dat componenten arbitraire props kunnen aannemen, waaronder primitieve waarden, React elementen of functies.

> Als je niet-UI functionaliteit wil herbruiken tussen componenten, raden we aan om ze te extraheren naar een aparte JavaScript module. De componenten kunnen die functie, object of class importeren en gebruiken, zonder het uit te breiden.

Reviewer: ### Voorgestelde wijzigingen
niet-UI functionaliteit => niet-UI-functionaliteit
tussen => voor meer
om ze te extraheren naar => om die er uit te halen naar
kunnen die => kunnen de
zonder het => zonder hem

```suggestion
Als je niet-UI-functionaliteit wil herbruiken voor meer componenten, raden we aan om die er uit te halen naar een aparte JavaScript module. De componenten kunnen de functie, object of class importeren en gebruiken, zonder hem uit te breiden.
```

### De tekst wordt dan
Als je niet-UI-functionaliteit wil herbruiken voor meer componenten, raden we aan om die er uit te halen naar een aparte JavaScript module. De componenten kunnen de functie, object of class importeren en gebruiken, zonder hem uit te breiden.

### Oorspronkelijk
Als je niet-UI functionaliteit wil herbruiken tussen componenten, raden we aan om ze te extraheren naar een aparte JavaScript module. De componenten kunnen die functie, object of class importeren en gebruiken, zonder het uit te breiden.
